### ISSUE TITLE
Integrate improvements from `resvg` crate

### ISSUE THREAD

**Issue by Shnatsel** [fschutt/fastblur#14]
`resvg` crate now includes [a modified copy](https://github.com/RazrFalcon/resvg/commit/695c0680fd32388015f16ef5c1e92669c41ab543#diff-ce811c080df082862d76f31b16c2e560R557) of the blur algorithm in this crate. Most notably it includes alpha channel support as well RGB/BRG/etc, but also fixes #13.

**Comment by fschutt** [comment]
Yeah, ideally this crate should be generic across array sizes, which is currently not stable in Rust.

**Comment by rljacobson** [comment]
Const generics are now in stable, so maybe it's time for another look.

Also useful would be an algorithm that operates on a suitable data type that has a dynamically determined number of dimensions. Surely someone has made a crate for such a data type.
